### PR TITLE
fix(octane): hold controlled inputs through a suspended transition

### PR DESCRIPTION
**Pull request by trueadm**
Follow-up to #379, which held plain bindings through a suspended transition but
left controlled form state out.

## Summary

A controlled input inside a held boundary still showed its new value while
everything around it held. It is now held too.

## Why it needed more than the node

A controlled input keeps three things in step:

1. the live DOM property (`value` / `checked` / `selected`),
2. the `default*` mirror that `form.reset()`, SSR output and the differential
   byte-compares are aligned against,
3. a per-element `$$ctrl` record of what was last projected.

Restoring only the node would have been worse than not restoring it. The record
would still believe the new value had landed, so re-projecting that same value on
resume would be skipped as unchanged and the input would stay behind
permanently. All three now go into the log together.

`<select>` needed one extra step: the selection lives on the options, not on the
select, so `projectSelectValue` records each option's `selected` (and
`defaultSelected` when it stamps the mount-time baseline) before it walks them.

## Changes

- `journalObjectOnce` factored out of `journalBag`, so the `$$ctrl` record is
  snapshotted by the same once-per-window mechanism as a compiled binding bag.
- `journalControlled` records the property, its `default*` mirror and the record.
  Called from `setValue`, `setChecked` and `setCheckedCheckable` once the element
  is armed and before the write touches anything.
- `setSelectValue` records the record; `projectSelectValue` records the options.
- `JOURNAL_PROP` reinstated for property writes.

## Validation

- [x] `pnpm test` — see the note below
- [x] `pnpm typecheck` — clean
- [x] `pnpm format:check` — clean
- [x] New regression test in `transitions.test.ts`: a controlled text input and
      checkbox beside a suspending sibling. Asserts both hold at step 0 with
      their `default*` mirrors, and — the part that catches a node-only restore —
      that both reach step 1 once the promise settles. Watched failing before the
      fix.

**On the full suite:** four runs on this branch gave 1, 2, 0 and 0 failures, all
confined to `hydration/deferred-hydration.test.ts` under `octane-prod`, which
passes in isolation both here and on `main`. A run of clean `main` in the same
environment gave 3 failures across 2 files. The flakiness is environmental
(timing-sensitive hydration tests under full-suite parallelism), pre-existing,
and not introduced here — but it is real and I would rather flag it than round it
to "all green".

No performance measurement: the added work is a `TRANSITION_JOURNAL !== null`
comparison on the controlled-write paths, which are not hot (one per controlled
binding per render) and are already doing property reads and dev diagnostics.

## Scope note — the two remaining gaps are one project

#379 listed three things a transition could still change early. This closes the
controlled-input one. The other two turned out to be the same problem, and I have
corrected `SUSPENSE_DIVERGENCE.md` #4 accordingly:

- **Content outside a suspended boundary.** Reverting it strands it. The reveal
  path re-renders the try block only, so a restored bag outside the boundary is
  never re-patched and the content stays on the old value for good. The hold
  would have to record the block the transition originated from and re-render
  that instead.
- **Structural changes above the boundary.** Moves and inserts journal fine
  (`node`, `parent`, `nextSibling`). Removals do not: the reconciler disposes
  blocks, running user cleanups and discarding hook state, and a cleanup cannot
  be un-run. Those disposals have to be collected during the render and executed
  only on commit.

Both need the transition render to become a deferred commit unit, along with
capture-and-splice for the rolled-back region's effects so they do not run
against DOM that was reverted underneath them. That is one piece of work, not two
follow-ups.

---

> [!NOTE]
> **Medium Risk**
> Touches hot controlled-input commit paths in `runtime.ts` behind a journal guard, with subtle radio/select edge cases; scope is bounded to transition suspend undo and regression tests.
> 
> **Overview**
> Extends the transition **binding journal** so controlled **`value`**, **`checked`**, and **`selected`** stay on the old screen when a Suspense boundary holds, not just text/attributes/bags.
> 
> Adds **`JOURNAL_PROP`** and **`journalControlled`** / **`journalObjectOnce`** to snapshot the live DOM property, its **`default*`** mirror, the per-element **`$$ctrl`** projection record, and (for **`<select>`**) each option’s **`selected`** / **`defaultSelected`** before writes. **`setValue`**, **`setChecked`**, **`setCheckedCheckable`**, **`setSelectValue`**, and **`projectSelectValue`** call into the journal when **`TRANSITION_JOURNAL`** is open; radio groups also journal the **cousin** the platform clears when another member is checked.
> 
> Docs and **`SUSPENSE_DIVERGENCE.md`** now treat controlled form state as closed and fold the remaining “outside boundary” / structural gaps into **one deferred-commit** limitation. New **`transitions.test.ts`** coverage covers controlled text/checkbox and radio groups beside a suspending sibling.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b06e54a3daf9501ec3a4cc3945b545393c106a79. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/octane/src/runtime.ts [hunk 1]
```diff
@@ -12783,7 +12821,9 @@ export function setChecked(el: Element, value: unknown): void {
  * and event restoration contract, but cannot need text-composition listeners.
  */
 export function setCheckedCheckable(el: Element, value: unknown): void {
-	setCheckedState(el as HTMLInputElement, value, armControlledBase(el));
+	const ctrl = armControlledBase(el);
+	if (TRANSITION_JOURNAL !== null) journalControlled(el, 'checked', 'defaultChecked');
+	setCheckedState(el as HTMLInputElement, value, ctrl);
```
[the remark this answers is not in the document]
Comment (trueadm) [reply packages/octane/src/runtime.ts:12865]: Confirmed and fixed in `b06e54a3`. This one was real.

Reproduced the mechanism directly before touching anything — two same-name radios, `b` checked, then the sequence the journal actually performs:

```
start          a=false b=true
after a=true   a=true  b=false     <- platform cleared b as a side effect
after undo     a=false b=false     <- b should be true
```

`b`'s own binding runs *after* `a`'s write, so the state it records is already the cleared one, and an uncontrolled cousin never records at all. Rollback then leaves the group with nothing selected.

Fixed by recording the cousin before the write, the same way `projectSelectValue` records every option. One refinement over journaling the whole group: only a cousin that is *currently checked* can be cleared by the write, and a well-formed group has at most one — so `journalRadioCousins` scans the group but adds at most one entry rather than one per radio.

Group scope mirrors `restoreRadioCousins` (same non-empty name, same form owner when there is one). Re-recording a cousin across several writes in one window is harmless: the replay runs newest-first, so the earliest recorded value is the one left standing.

Pinned with a regression test in `transitions.test.ts` — a two-radio group beside a suspending sibling, asserting the group returns to `b` rather than to nothing, and reaches `a` once the promise settles. Watched it fail before the fix with exactly the symptom you described.

Separately, writing this caught a bug of my own: my first cut pushed 5 values into a log with a 4-slot stride, which would have desynchronised every entry after it. Fixed, and I've verified all eight push sites are 4-wide.
